### ISSUE TITLE
Hugging Face Trainer + Accelerate Notebook Launcher?

### ISSUE THREAD

**Issue by haochen806** [huggingface/accelerate#1956]
### System Info

```Shell
From this https://github.com/huggingface/accelerate/issues/144, I would assume trainer works seemlessly with accelerate code now.
I want to confirm if it works seemlessly with the notebook launcher provided by accelerate library.
```

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [ ] My own task or dataset (give details below)

### Reproduction

```
def training_loop():
    trainer.train()
notebook_launcher(training_loop)
```

### Expected behavior

Work as expected.

**Comment by LysandreJik** [comment]
cc @pacman100

**Comment by muellerzr** [comment]
Yes, the notebook launcher works with any pytorch code with the same rules applied: cuda can’t be initialized outside the training loop function you make.
